### PR TITLE
Enable shareable configuration feature and update init logic

### LINKED ISSUE #1258: Update share configuration according to the new design and remove it from the feature flag

### PR DESCRIPTION
**Pull request by dreampulse**
Merge #1256 first! Change base to `develop`!

resolves #1258

### CODE REVIEW

File: src/app/container/main-app.js [hunk 1]
```diff
@@ -19,9 +17,37 @@ import ConfigurationPage from './configuration-page'
 
 import Modal from './modal'
 
-const MainApp = ({initDone}) => {
+const Empty = ({onMount}) => {
```
Comment (webholics) [inline src/app/container/main-app.js]: this name is weird

File: src/app/container/main-app.js [hunk 2]
```diff
@@ -19,9 +17,37 @@ import ConfigurationPage from './configuration-page'
...
+  }
+
   if (!initDone) {
-    return null
+    return (
+      <Switch>
+        <Route
+          path="/configuration/:id"
+          exact
+          component={() => <Empty onMount={init({restoreSessionEnabled: false})} />}
```
Comment (webholics) [inline src/app/container/main-app.js]: from the react router doc: "When using an inline function for inline rendering, use the render or the children prop (below)"

https://reacttraining.com/react-router/web/api/Route/render-func

File: src/app/container/main-app.js [hunk 3]
```diff
@@ -19,9 +17,37 @@ import ConfigurationPage from './configuration-page'
 
 import Modal from './modal'
 
-const MainApp = ({initDone}) => {
+const Empty = ({onMount}) => {
+  useEffect(() => {
```
Comment (webholics) [inline src/app/container/main-app.js]: would prefer to use `lifecycle` and componentDidMount

File: src/app/container/upload-page.js [hunk 4]
```diff
@@ -253,19 +270,6 @@ const UploadPage = ({
                 unconfiguredConfigIds.length
               } files)`}
             />
-            <Paragraph>
```
Comment (webholics) [inline src/app/container/upload-page.js]: der paragraph import oben muss raus

**Comment by dreampulse** [comment]
TODO: Text muss noch angepasst werden

**Pull request merged**